services/osparc-gateway-server/tests/unit/test_utils.py

```diff
@@ -36,46 +42,57 @@ def minimal_config(monkeypatch):
     )
 
 
-@pytest.fixture
+@pytest.fixture()
 async def docker_service(
-    async_docker_client: aiodocker.Docker,
-) -> AsyncIterator[Dict[str, Any]]:
-    TaskTemplate = {
-        "ContainerSpec": {
-            "Image": "redis",
-        },
-    }
-    service = await async_docker_client.services.create(
-        task_template=TaskTemplate, name="my_service"
-    )
-    assert service
-    print(f"--> created docker service {service}")
-    inspected_service = await async_docker_client.services.inspect(service["ID"])
-    print(f"--> service inspected returned {inspected_service}")
+    docker_swarm, async_docker_client: aiodocker.Docker, faker: Faker
+) -> AsyncIterator[Callable[[Dict[str, str]], Awaitable[Dict[str, Any]]]]:
+    created_services = []
+
+    async def _creator(labels: Dict[str, str]) -> Dict[str, Any]:
+        service = await async_docker_client.services.create(
+            task_template={
+                "ContainerSpec": {"Image": "busybox", "Command": ["sleep", "10000"]}
+            },
+            name=faker.pystr(),
+            labels=labels,
+        )
+        assert service
+        created_services.append(service)
+        print(f"--> created docker service {service}")
+        inspected_service = await async_docker_client.services.inspect(service["ID"])
+        print(f"--> service inspected returned {inspected_service}")
+        return inspected_service
+
+    yield _creator
 
-    yield inspected_service
-    # cleanup
-    await async_docker_client.services.delete(service["ID"])
+    await asyncio.gather(
+        *[async_docker_client.services.delete(s["ID"]) for s in created_services]
+    )
 
 
 @pytest.fixture
-async def running_service(
-    async_docker_client: aiodocker.Docker, docker_service: Dict[str, Any]
-) -> Dict[str, Any]:
-    async for attempt in AsyncRetrying(
-        reraise=True, wait=wait_fixed(1), stop=stop_after_delay(60)
-    ):
-        with attempt:
-            tasks = await async_docker_client.tasks.list(
-                filters={"service": f"{docker_service['Spec']['Name']}"}
-            )
-            task_states = [task["Status"]["State"] for task in tasks]
-            num_running = sum(current == "running" for current in task_states)
-            print(f"--> service task states {task_states=}")
-            assert num_running == 1
-            print(f"--> service {docker_service['Spec']['Name']} is running now")
-            return docker_service
-    raise AssertionError(f"service {docker_service=} could not start")
+def running_service(
```

Comment:
MINOR: this is callable, why don't you use a verb in it since it denotes a creation action: e.g.  ``create_running_service`` ..? It reads much better
Same with ``docker_service``.
I find it confusing to use nouns for actions

Reply:
ok, this is due to evolutive coding... will refactor